### PR TITLE
feat: add list project datasets API

### PR DESCRIPTION
**Pull request by bhudnell**
Issue #, if available:

Description of changes:
- Added ListDataSetsForProject API

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-reference/src/services/dataSetService.ts [hunk 1]
```diff
@@ -164,6 +208,14 @@ export class DataSetService implements DataSetPlugin {
       ['READ', 'UPDATE', 'DELETE']
     );
 
+    await this._addAuthZPermissionsForDataset(
+      request.authenticatedUser,
+      SwbAuthZSubject.SWB_DATASET,
+      dataset.id!,
+      [projectResearcher],
+      ['READ', 'UPDATE']
```
Comment (bhudnell) [inline solutions/swb-reference/src/services/dataSetService.ts:216]:
```suggestion
      ['READ']
```

**Review by kpark277 — APPROVED** [review]

File: solutions/swb-app/src/dataSetRoutes.ts [hunk 2]
```diff
@@ -156,6 +155,25 @@ export function setUpDSRoutes(router: Router, dataSetService: DataSetPlugin): vo
     })
   );
 
+  // List project dataSets
+  router.get(
+    '/projects/:projectId/datasets',
+    wrapAsync(async (req: Request, res: Response) => {
+      const authenticatedUser = validateAndParse<AuthenticatedUser>(AuthenticatedUserParser, res.locals.user);
+      const pageSize = toNumber(req.query.pageSize) || DEFAULT_API_PAGE_SIZE;
```
Comment (htztomic) [inline solutions/swb-app/src/dataSetRoutes.ts:163]: Limit to a MAX_API_PAGE_SIZE

File: solutions/swb-reference/integration-tests/tests/isolated/projects/listDataSetsForProject.test.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  describe('negative tests', () => {
+    test('IT Admin cannot view any datasets, so the call will return an empty list', async () => {
+      const { data } = await itAdminSession.resources.projects.project(project1Id).dataSets().get();
+
+      expect(data.data).toStrictEqual([]);
+    });
+
+    test('Project Admin from project 2 cannot list datasets for project 1', async () => {
+      try {
+        await pa2Session.resources.projects.project(project1Id).dataSets().get();
```
Comment (htztomic) [inline solutions/swb-reference/integration-tests/tests/isolated/projects/listDataSetsForProject.test.ts:45]: Possible negative case: User belongs to a group with access to dataset 1 through project 1 but is trying to access dataset 1 through project 2?

File: solutions/swb-app/src/dataSetRoutes.ts (hunk 2 above)
Comment (bhudnell) [reply solutions/swb-app/src/dataSetRoutes.ts:163]: done

File: solutions/swb-reference/integration-tests/tests/isolated/projects/listDataSetsForProject.test.ts (hunk 3 above)
Comment (bhudnell) [reply solutions/swb-reference/integration-tests/tests/isolated/projects/listDataSetsForProject.test.ts:45]: Will handle this test in another PR.

File: solutions/swb-app/src/dataSetRoutes.ts [hunk 4]
```diff
@@ -151,11 +151,38 @@ export function setUpDSRoutes(router: Router, dataSetService: DataSetPlugin): vo
...
+      const paginationToken = req.query.paginationToken?.toString();
+      const projectId = req.params.projectId;
+
+      if (pageSize < 1 || pageSize > MAX_API_PAGE_SIZE) {
```
Comment (htztomic) [inline solutions/swb-app/src/dataSetRoutes.ts:172]: Should be idempotent, if they called a larger number than MAX, just floor it to the MAX

Comment (kpark277) [reply solutions/swb-app/src/dataSetRoutes.ts:172]: I prefer erroring so that this doesn't lead to unexpected behavior.

**Pull request merged**